Sounds good. After running shellcheck, no complains were found on the proposed code.

There are some minor "double quote" warnings on code corresponding to graphite.

For reference, shellcheck output:
```
[Line 8:](javascript:setPosition(8, 14))
        echo $1 | tr '.' '_'
             ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.

Did you mean: ([apply this](javascript:applyFixIndex([0])), apply [all SC2086](javascript:applyFixCode(2086)))
        echo "$1" | tr '.' '_'
 
[Line 11:](javascript:setPosition(11, 28))
gprefix="nfs.$(graphiteify $(hostname -f)).clients"
                           ^-- [SC2046](https://www.shellcheck.net/wiki/SC2046) (warning): Quote this to prevent word splitting.
 
[Line 83:](javascript:setPosition(83, 36))
                host=$(graphiteify $(dig +short -x $ipaddr | sed 's/\.$//'))
                                   ^-- [SC2046](https://www.shellcheck.net/wiki/SC2046) (warning): Quote this to prevent word splitting.
                                                   ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.

Did you mean: ([apply this](javascript:applyFixIndex([3])), apply [all SC2086](javascript:applyFixCode(2086)))
                host=$(graphiteify $(dig +short -x "$ipaddr" | sed 's/\.$//'))
 
[Line 87:](javascript:setPosition(87, 48))
                        ip_under=$(graphiteify $ipaddr)
                                               ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.

Did you mean: ([apply this](javascript:applyFixIndex([4])), apply [all SC2086](javascript:applyFixCode(2086)))
                        ip_under=$(graphiteify "$ipaddr")
```

--- a/bpftrace/nfsd-clients.bt
+++ b/bpftrace/nfsd-clients.bt
@@ -14,9 +14,9 @@ struct nfsd4_op {
     u32 opnum;
 };
 
-kprobe:nfsd4_encode_operation {
-    $res = (struct nfsd4_compoundres *)arg0;
-    $op = (struct nfsd4_op *)arg1;
+fentry:nfsd4_encode_operation {
+    $res = args.resp;
+    $op = args.op;
 
     // Pointer to the request struct
     $rqstp = (struct svc_rqst *)$res->rqstp;

--- a/shell-scripts/collect-bpf-nfsd.sh
+++ b/shell-scripts/collect-bpf-nfsd.sh
@@ -5,10 +5,10 @@ bpfscript='/usr/local/sbin/bpf-nfsd.bt'
 
 # only used for graphite
 graphiteify() {
-	echo $1 | tr '.' '_'
+	echo "$1" | tr '.' '_'
 }
 gtmpfile='/tmp/graphite-nfsclients'
-gprefix="nfs.$(graphiteify $(hostname -f)).clients"
+gprefix="nfs.$(graphiteify "$(hostname -f)").clients"
 
 HELP="Usage: collect-bpf-nfsd.sh [OPTIONS]
 
@@ -80,11 +80,11 @@ graphite() {
 
 	# try to resolve ip addresses
 	for ipaddr in $(awk '{print $4}' "$gtmpfile" | sort | uniq); do
-		host=$(graphiteify $(dig +short -x $ipaddr | sed 's/\.$//'))
+		host=$(graphiteify "$(dig +short -x "$ipaddr" | sed 's/\.$//')")
 		if [ "$host" != "" ]; then
 			sed -i "s/ ${ipaddr}$/ ${host}/g" "$gtmpfile"
 		else
-			ip_under=$(graphiteify $ipaddr)
+			ip_under=$(graphiteify "$ipaddr")
 			sed -i "s/ ${ipaddr}$/ ${ip_under}/g" "$gtmpfile"
 		fi
 	done

